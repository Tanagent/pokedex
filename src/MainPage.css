.btn-container {
    width: 100%;
    display: flex;
    justify-content: center;
    margin: 5px;
}

.all-btn {
    background-color: #FFCB05;
    height: 30px;
	width: 90px;
    font-size: large;
    margin: 5px;
}

.bag-btn {
    background-color: #3D7DCA;
    height: 30px;
	width: 90px;
    font-size: large;
    margin: 5px;
}

.save-btn {
    background-color: green;
    height: 30px;
	width: 90px;
    font-size: large;
    margin: 5px;
}

.clear-btn {
    background-color: rgb(180, 48, 180);
    height: 30px;
	width: 90px;
    font-size: large;
    margin: 5px;
}

.search-field {
    margin: auto;
    display: block;
    width: 500px;
    padding: 5px;
    margin-bottom: 10px;
}

.grid-container {
    display: flex;
    justify-content: center;
    display: grid;
    grid-template-columns: repeat(auto-fill, 400px);
    gap: 50px;
}