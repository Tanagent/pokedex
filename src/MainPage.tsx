import React, { useState, useEffect, useContext } from 'react';
import PokemonContainer from './PokemonContainer';
import { PokemonContext } from './PokemonContext';
import './MainPage.css';
import axios from 'axios';

interface IPokemon {
    name: string
}

const MainPage = () => {
    const { bag } = useContext(PokemonContext);
    const [showAll, setShowAll] = useState(true);
    const [pokemon, setPokemon] = useState<Array<IPokemon>>([]);
    const [searchValue, setSearchValue] = useState('');

    useEffect(() => {
        if (pokemon === undefined || pokemon.length === 0) {
            let mounted = true;
            const fetchPokemon = async () => {
                for(let i = 1; i <= 151; i++) {
                    await getPokemon(i);
                }
            }

            const getPokemon = async (id: number) => {
                const url = `https://pokeapi.co/api/v2/pokemon/${id}`
                const pokemon = await axios.get(url);
                if(mounted) 
                    setPokemon(item => [...item, pokemon.data]);
            }
            
            fetchPokemon();
            
            return () => {
                mounted = false;
            }
        }
    }, []);

    return (
        <div>
            <div className="button-container m-auto pb-5">
                <button className='all-btn' onClick={() => setShowAll(true)}>All</button><button className='bag-btn' onClick={() => setShowAll(false)}>Bag</button>
                {!showAll && <button className='save-btn'>Save</button>}
            </div>
            <input placeholder="Seach" type="text" value={searchValue} onChange={(event) => setSearchValue(event.target.value)}></input>
            {
                (showAll ? pokemon : bag).filter((pokemon: IPokemon) => 
                    searchValue === '' || pokemon.name.toLowerCase().includes(searchValue)
                ).map((pokemon: any, index: number) => (
                    <div key={index}>
                        {
                            showAll ? <PokemonContainer id={pokemon.id} name={pokemon.name} img={pokemon.sprites.front_default} /> :
                            <PokemonContainer id={pokemon.id} name={pokemon.name} img={pokemon.img} />
                        }
                    </div>

                ))
            }
        </div>
    );
}

export default MainPage;