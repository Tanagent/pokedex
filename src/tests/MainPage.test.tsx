import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount} from 'enzyme';
import MainPage from '../MainPage';
import MockAdapter from 'axios-mock-adapter';
import { act, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

const mockAxios = {
    get: jest.fn(() => Promise.resolve({ 
        data: {
            id: 132,
            name: "ditto",
            sprites: {
                front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
            }
        }
    }))
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MainPage />, div);
});

it('shallow renders without crashing', () => {
  shallow(<MainPage />);
});

const PokemonList = new Array(
    "Bulbasaur","Ivysaur","Venusaur","Charmander","Charmeleon","Charizard","Squirtle","Wartortle","Blastoise","Caterpie",
    "Metapod","Butterfree","Weedle","Kakuna","Beedrill","Pidgey","Pidgeotto","Pidgeot","Rattata","Raticate",
    "Spearow","Fearow","Ekans","Arbok","Pikachu","Raichu","Sandshrew","Sandslash","Nidoran","Nidorina",
    "Nidoqueen","Nidoran","Nidorino","Nidoking","Clefairy","Clefable","Vulpix","Ninetales","Jigglypuff","Wigglytuff",
    "Zubat","Golbat","Oddish","Gloom","Vileplume","Paras","Parasect","Venonat","Venomoth","Diglett",
    "Dugtrio","Meowth","Persian","Psyduck","Golduck","Mankey","Primeape","Growlithe","Arcanine","Poliwag",
    "Poliwhirl","Poliwrath","Abra","Kadabra","Alakazam","Machop","Machoke","Machamp","Bellsprout","Weepinbell",
    "Victreebel","Tentacool","Tentacruel","Geodude","Graveler","Golem","Ponyta","Rapidash","Slowpoke","Slowbro",
    "Magnemite","Magneton","Farfetch'd","Doduo","Dodrio","Seel","Dewgong","Grimer","Muk","Shellder",
    "Cloyster","Gastly","Haunter","Gengar","Onix","Drowzee","Hypno","Krabby","Kingler","Voltorb",
    "Electrode","Exeggcute","Exeggutor","Cubone","Marowak","Hitmonlee","Hitmonchan","Lickitung","Koffing","Weezing",
    "Rhyhorn","Rhydon","Chansey","Tangela","Kangaskhan","Horsea","Seadra","Goldeen","Seaking","Staryu","Starmie",
    "Mr. Mime","Scyther","Jynx","Electabuzz","Magmar","Pinsir","Tauros","Magikarp","Gyarados","Lapras","Ditto",
    "Eevee","Vaporeon","Jolteon","Flareon","Porygon","Omanyte","Omastar","Kabuto","Kabutops","Aerodactyl"
    ,"Snorlax","Articuno","Zapdos","Moltres","Dratini","Dragonair","Dragonite","Mewtwo","Mew");

describe('MainPage', () => {
    it('should display all 151 pokemon', async () => {
        mockAxios.get.mockImplementationOnce(() => Promise.resolve({
            data: {
                id: 132,
                name: "ditto",
                sprites: {
                    front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
                }
            }
        }))

        //const component = mount(<MainPage />);
        //console.log('comp', component);
        //const images = await MainPage();
        //console.log('images', images);

        // // mock axios promise
        // let wrapper = mount(<MainPage />);
        // await act(async () => {
        //     await mockAxios.get.mockImplementationOnce(() => Promise.resolve({
        //         data: {
        //             id: 132,
        //             name: "ditto",
        //             sprites: {
        //                 front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
        //             }
        //         }
        //     }));
        //     wrapper = mount(<MainPage />);
        // });
  
        // check the render output
        // wrapper.update();

        // const tree = renderer
        // .create(<MainPage />);
        // await Promise.resolve();
    })
})