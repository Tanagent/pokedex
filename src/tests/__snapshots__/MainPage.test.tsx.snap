// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should display all 151 pokemon 1`] = `
<div>
  <div
    className="button-container m-auto pb-5"
  >
    <button
      className="all-btn"
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="bag-btn"
      onClick={[Function]}
    >
      Bag
    </button>
  </div>
  <input
    onChange={[Function]}
    placeholder="Seach"
    type="text"
    value=""
  />
</div>
`;
